packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -419,8 +419,19 @@ export class CanvasManager implements CanvasManagerInterface {
 
       for (const item of this.windows) {
         const window = item.deref();
-        if (window) {
-          searchCanvas(window.document);
+        let _document: Document | false | undefined;
+
+        try {
+          _document = window && window.document;
+        } catch {
+          // Accesing `window.document` can throw a security error:
+          // "Failed to read a named property 'document' from 'Window': An
+          // attempt was made to break through the security policy of the user
+          // agent."
+        }
+
+        if (_document) {
+          searchCanvas(_document);
```

Comment:
```suggestion
        if (_document) {
          // This is not included in the `try` block above in case `searchCanvas()` throws
          searchCanvas(_document);
```